### ISSUE TITLE
Slack: Inout gateway between slack and irc or xmpp does not forward messages from slack

### ISSUE THREAD

**Issue by cfelder** [42wim/matterbridge#299]
### Which version of matterbridge are you using?
version: 1.4.0

### Please describe the expected behavior.

* matterbridge gateway which should forward messages (inout) between slack, irc and xmpp

### Please describe the actual behavior.

* To slack I get all messages (either from irc or xmpp) in channel bridge.
* I don't get the answers from slack (neither to irc nor to xmpp).

The debugging output contains three messages:

* hello from slack
* hello from irc
* hello from xmpp

although the message ``hello from slack`` is recognised it is not forwarded

```
time="2017-11-13T14:14:20+01:00" level=info msg="Enabling debug"
time="2017-11-13T14:14:20+01:00" level=info msg="Running version 1.4.0 "
time="2017-11-13T14:14:20+01:00" level=info msg="Starting bridge: irc.myirc "
time="2017-11-13T14:14:20+01:00" level=info msg="Connecting irc.opensuse.org:6667" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg=""CLIENT_INIT :irc.opensuse.org:6667"" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net NOTICE * :*** Looking up your hostname..."" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net NOTICE * :*** Checking Ident"" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net NOTICE * :*** Found your hostname"" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net NOTICE * :*** No Ident response"" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net CAP * LS :account-notify extended-join identify-msg multi-prefix sasl"" module=irc
time="2017-11-13T14:14:20+01:00" level=debug msg="":cherryh.freenode.net 433 * mattermost :Nickname is already in use."" module=irc
time="2017-11-13T14:14:21+01:00" level=debug msg="":cherryh.freenode.net CAP * ACK :account-notify extended-join multi-prefix "" module=irc
time="2017-11-13T14:14:21+01:00" level=debug msg="":cherryh.freenode.net 001 mattermost_ :Welcome to the freenode Internet Relay Chat Network mattermost_"" module=irc
time="2017-11-13T14:14:21+01:00" level=debug msg="Registering callbacks" module=irc
time="2017-11-13T14:14:21+01:00" level=debug msg=""CLIENT_GENERAL_UPDATED"" module=irc
time="2017-11-13T14:14:21+01:00" level=info msg="Connection succeeded" module=irc
time="2017-11-13T14:14:21+01:00" level=info msg="irc.myirc: joining #cfelder (#cfelderirc.myirc)"
time="2017-11-13T14:14:21+01:00" level=info msg="Starting bridge: slack.myslack "
time="2017-11-13T14:14:21+01:00" level=info msg="Connecting using token (sending and receiving)" module=slack
time="2017-11-13T14:14:21+01:00" level=info msg="slack.myslack: joining bridge (bridgeslack.myslack)"
time="2017-11-13T14:14:21+01:00" level=debug msg="Choosing token based receiving" module=slack
time="2017-11-13T14:14:21+01:00" level=info msg="Starting bridge: xmpp.myxmpp "
time="2017-11-13T14:14:21+01:00" level=info msg="Connecting example.com:5222" module=xmpp
<?xml version='1.0'?><stream:stream xmlns:stream='http://etherx.jabber.org/streams' xmlns='jabber:client' from='example.com' version='1.0' id='85wqu72qll4m4nydubgfy6ow8zxcpbrx6jzuw97b' xmlns:ack='http://www.xmpp.org/extensions/xep-0198.html#ns'>
<stream:features xmlns:stream='http://etherx.jabber.org/streams'><starttls xmlns='urn:ietf:params:xml:ns:xmpp-tls'><required/></starttls><address xmlns='http://affinix.com/jabber/address'>::ffff:172.25.80.13</address><auth xmlns='http://jabber.org/features/iq-auth'/></stream:features>
<proceed xmlns='urn:ietf:params:xml:ns:xmpp-tls'/>
<?xml version='1.0'?><stream:stream xmlns:stream='http://etherx.jabber.org/streams' xmlns='jabber:client' from='example.com' version='1.0' id='9vds9eeaf95313xbel630c4s607kedpsxq6z8s1v' xmlns:ack='http://www.xmpp.org/extensions/xep-0198.html#ns'>
<stream:features xmlns:stream='http://etherx.jabber.org/streams'><address xmlns='http://affinix.com/jabber/address'>::ffff:172.25.80.13</address><mechanisms xmlns='urn:ietf:params:xml:ns:xmpp-sasl'><mechanism>GSSAPI</mechanism><mechanism>GSSAPI</mechanism><mechanism>DIGEST-MD5</mechanism><mechanism>DIGEST-MD5</mechanism><mechanism>PLAIN</mechanism></mechanisms><auth xmlns='http://jabber.org/features/iq-auth'/></stream:features>
<challenge xmlns='urn:ietf:params:xml:ns:xmpp-sasl'>bm9uY2U9IjA4b1N6RkNrQjhFZEN4cUFUa1k4RG5HZXJFNDJGeHhPNnJ0VGJkL3hId3hISGFSYSIscmVhbG09ImlmZmltLmlmZi5rZmEtanVlbGljaC5kZSIsY2hhcnNldD11dGYtOCxxb3A9ImF1dGgiLGFsZ29yaXRobT1tZDUtc2Vzcw==</challenge>
<challenge xmlns='urn:ietf:params:xml:ns:xmpp-sasl'>cnNwYXV0aD0yMDYxMDVkM2VmMzljYTUwZGZhNzA0ZGVlNzdiNDBmMA==</challenge>
<success xmlns='urn:ietf:params:xml:ns:xmpp-sasl'/>
<?xml version='1.0'?><stream:stream xmlns:stream='http://etherx.jabber.org/streams' xmlns='jabber:client' from='example.com' version='1.0' id='u6n4sj67agnwkp77sygyfitl7er2ubfv1f5m0rpr' xmlns:ack='http://www.xmpp.org/extensions/xep-0198.html#ns'>
<stream:features xmlns:stream='http://etherx.jabber.org/streams'><ack:ack/><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'><required/></bind><unbind xmlns='urn:ietf:params:xml:ns:xmpp-bind'/><ver xmlns='urn:xmpp:features:rosterver'/></stream:features>
<iq xmlns='jabber:client' id='13f5740aa88633d2' type='result'><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'><jid>felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0</jid></bind></iq>
time="2017-11-13T14:14:22+01:00" level=info msg="Connection succeeded" module=xmpp
time="2017-11-13T14:14:22+01:00" level=info msg="xmpp.myxmpp: joining bridge (bridgexmpp.myxmpp)"
time="2017-11-13T14:14:22+01:00" level=info msg="Gateway(s) started succesfully. Now relaying messages"
<iq xmlns='jabber:client' id='13f5740aa88633d2' type='result' to='example.com'/>
time="2017-11-13T14:14:22+01:00" level=debug msg="Start listening for Slack messages" module=slack
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='bridge@rooms.example.com/XXXUSER'><priority>1</priority><c xmlns='http://jabber.org/protocol/caps' node='http://pidgin.im/' hash='sha-1' ver='lV6i//bt2U8Rm0REcX8h4F3Nk3M=' ext='voice-v1 camera-v1 video-v1'/><x xmlns='http://jabber.org/protocol/muc'/><x xmlns='http://jabber.org/protocol/muc#user'><item role='participant' affiliation='none' jid='XXXUSER@example.com/6b5065d5f938d4a1c6d92ef9f55767c8ce4e0001'/></x></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='bridge@rooms.example.com/felder'><x xmlns='http://jabber.org/protocol/muc'><history maxstanzas='25'/></x>
<x xmlns='http://jabber.org/protocol/muc#user'><item role='moderator' affiliation='owner' jid='felder@example.com/localhost'/></x></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='bridge@rooms.example.com/matterbridge'>
<x xmlns='http://jabber.org/protocol/muc'><history maxchars='0'/></x>
<x xmlns='http://jabber.org/protocol/muc#user'><item role='moderator' affiliation='owner' jid='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0'/><status code='110'/></x></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='bridge@rooms.example.com/matterbridge'>
<x xmlns='http://jabber.org/protocol/muc'><history maxchars='0'/></x>
<x xmlns='http://jabber.org/protocol/muc#user'><item role='moderator' affiliation='owner' jid='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0'/><status code='100'/><status code='110'/></x></presence>
<presence xmlns='jabber:client' from='XXXUSER@jabber.de' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' type='unavailable'/>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/190849b3a77e52ef25a59f667a6e0d247355d6f7'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/190849b3a77e52ef25a59f667a6e0d247355d6f7' stamp='2017-11-13T13:03:47Z'/><priority>1</priority><c xmlns='http://jabber.org/protocol/caps' ext='voice-v1 camera-v1 video-v1' ver='lV6i//bt2U8Rm0REcX8h4F3Nk3M=' hash='sha-1' node='http://pidgin.im/'/><x xmlns='vcard-temp:x:update'><photo/></x></presence>
<iq xmlns='jabber:client' id='iChat_E7D8DBEA' type='get' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@jabber.org/exhost449'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/20e3f530489b640b52b533d2e70c73232d108582' id='422ed9b0-26d0-cd4b-a3f7-1cb4a56d7217' xml:lang='en'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/20e3f530489b640b52b533d2e70c73232d108582' stamp='2017-11-13T13:09:54Z'/></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/work_example'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/work_example' stamp='2017-11-13T10:20:01Z'/><show>away</show><status>I&apos;m not here right now</status><query xmlns='jabber:iq:last' seconds='590662'/><c xmlns='http://jabber.org/protocol/caps' ext='voice-v1 camera-v1 video-v1' ver='lV6i//bt2U8Rm0REcX8h4F3Nk3M=' hash='sha-1' node='http://pidgin.im/'/><x xmlns='vcard-temp:x:update'><photo>5105f8b88e02b5e4c2d37ce890b8fec003f5d4b7</photo></x></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhostdev'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhostdev' stamp='2017-11-13T12:54:30Z'/><priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'><photo>D1697AFD38F4037AB1A64D52D33B4F45D2D62356</photo>
</x>
</presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhost449'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhost449' stamp='2017-11-13T12:35:42Z'/><priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'><photo>FC40A5D960F3CC3C791BB1D38BE230AE4718F035</photo>
</x>
</presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhost023'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhost023' stamp='2017-11-13T13:13:07Z'/><show>xa</show>
<priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'><photo>82E6864E78D4FF1338DD4F000AC6D66DAB21A39E</photo>
</x>
<x xmlns='http://www.apple.com/xmpp/idle'><idle-since>2017-11-13T13:03:06Z</idle-since>
</x>
</presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhost563'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhost563' stamp='2017-11-13T13:11:01Z'/><priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'/></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/6b5065d5f938d4a1c6d92ef9f55767c8ce4e0001'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/6b5065d5f938d4a1c6d92ef9f55767c8ce4e0001' stamp='2017-11-13T13:00:13Z'/><priority>1</priority><c xmlns='http://jabber.org/protocol/caps' ext='voice-v1 camera-v1 video-v1' ver='lV6i//bt2U8Rm0REcX8h4F3Nk3M=' hash='sha-1' node='http://pidgin.im/'/><x xmlns='vcard-temp:x:update'><photo>0e6bc5485115f8297c749c00f249bfb302ae76f4</photo></x></presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhost792'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhost792' stamp='2017-11-13T12:44:11Z'/><priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'><photo>98B91E75D7142E292EE3AFDF8FC207AD88A6EE13</photo>
</x>
</presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='XXXUSER@example.com/exhostaopc67 (2)'><delay xmlns='urn:xmpp:delay' from='XXXUSER@example.com/exhostaopc67 (2)' stamp='2017-11-13T13:11:24Z'/><show>xa</show>
<priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'/><x xmlns='http://www.apple.com/xmpp/idle'><idle-since>2017-11-13T13:01:23Z</idle-since>
</x>
</presence>
<presence xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='felder@example.com/localhost'><delay xmlns='urn:xmpp:delay' from='felder@example.com/localhost' stamp='2017-11-13T12:52:26Z'/><priority>0</priority>
<c xmlns='http://jabber.org/protocol/caps' ext='ice' ver='1000' node='http://www.apple.com/ichat/caps'/><x xmlns='http://jabber.org/protocol/tune'/><x xmlns='vcard-temp:x:update'><photo>489E6D95AF8B74B1ABF8B933AD1C17A05CE06C6A</photo>
</x>
</presence>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhostdev' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_2777A1EE' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhost449' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_55498C62' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhost023' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_4E512C16' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhost563' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_AAB37AC4' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhost792' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_3259F8CA' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
<iq xmlns='jabber:client' from='XXXUSER@example.com/exhostaopc67 (2)' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' id='iChat_489DB4CB' type='get'><query xmlns='http://jabber.org/protocol/disco#info'/></iq>
time="2017-11-13T14:14:27+01:00" level=debug msg="handle girc.Event{Source:(*girc.Source)(0xc420444660), Tags:girc.Tags(nil), Command:"JOIN", Params:[]string{"#cfelder", "*"}, Trailing:"mattermost", EmptyTrailing:false, Sensitive:false}" module=irc
time="2017-11-13T14:14:35+01:00" level=debug msg="Receiving from slackclient &slack.MessageEvent{Msg:slack.Msg{Type:"message", Channel:"C80ARRY15", User:"U24BVTPKP", Text:"hello from slack", Timestamp:"1510578875.000274", ThreadTimestamp:"", IsStarred:false, PinnedTo:[]string(nil), Attachments:[]slack.Attachment(nil), Edited:(*slack.Edited)(nil), SubType:"", Hidden:false, DeletedTimestamp:"", EventTimestamp:"", BotID:"", Username:"", Icons:(*slack.Icon)(nil), Inviter:"", Topic:"", Purpose:"", Name:"", OldName:"", Members:[]string(nil), ReplyCount:0, Replies:[]slack.Reply(nil), ParentUserId:"", File:(*slack.File)(nil), Upload:false, Comment:(*slack.Comment)(nil), ItemType:"", ReplyTo:0, Team:"T24BUPZC1", Reactions:[]slack.ItemReaction(nil)}, SubMessage:(*slack.Msg)(nil)}" module=slack
time="2017-11-13T14:14:36+01:00" level=debug msg="Sending message from cfelder on slack.myslack to gateway" module=slack
time="2017-11-13T14:14:47+01:00" level=debug msg="handlePrivMsg() cfelder hello from irc girc.Event{Source:(*girc.Source)(0xc420453260), Tags:girc.Tags(nil), Command:"PRIVMSG", Params:[]string{"#cfelder"}, Trailing:"hello from irc", EmptyTrailing:false, Sensitive:false}" module=irc
time="2017-11-13T14:14:47+01:00" level=debug msg="detected ISO-8859-1 confidence 20" module=irc
time="2017-11-13T14:14:47+01:00" level=debug msg="Sending message from #cfelder on irc.myirc to gateway" module=irc
time="2017-11-13T14:14:47+01:00" level=debug msg="Sending config.Message{Text:"hello from irc", Channel:"#cfelder", Username:"cfelder", UserID:"~cfelder@nat.example.com", Avatar:"", Account:"irc.myirc", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a8491c29c1aaf, ext:26405764907, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)} from irc.myirc (#cfelder) to slack.myslack (bridge)"
time="2017-11-13T14:14:47+01:00" level=debug msg="Receiving config.Message{Text:"hello from irc", Channel:"bridge", Username:"[irc] <cfelder> ", UserID:"~cfelder@nat.example.com", Avatar:"", Account:"irc.myirc", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a8491c29c1aaf, ext:26405764907, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)}" module=slack
time="2017-11-13T14:14:47+01:00" level=debug msg="Receiving from slackclient &slack.MessageEvent{Msg:slack.Msg{Type:"message", Channel:"C80ARRY15", User:"", Text:"hello from irc", Timestamp:"1510578887.000120", ThreadTimestamp:"", IsStarred:false, PinnedTo:[]string(nil), Attachments:[]slack.Attachment{slack.Attachment{Color:"", Fallback:"", CallbackID:"matterbridge", AuthorName:"", AuthorSubname:"", AuthorLink:"", AuthorIcon:"", Title:"", TitleLink:"", Pretext:"", Text:"", ImageURL:"", ThumbURL:"", Fields:[]slack.AttachmentField(nil), Actions:[]slack.AttachmentAction(nil), MarkdownIn:[]string(nil), Footer:"", FooterIcon:"", Ts:""}}, Edited:(*slack.Edited)(nil), SubType:"bot_message", Hidden:false, DeletedTimestamp:"", EventTimestamp:"1510578887.000120", BotID:"B7Z4X73PV", Username:"[irc] <cfelder>", Icons:(*slack.Icon)(nil), Inviter:"", Topic:"", Purpose:"", Name:"", OldName:"", Members:[]string(nil), ReplyCount:0, Replies:[]slack.Reply(nil), ParentUserId:"", File:(*slack.File)(nil), Upload:false, Comment:(*slack.Comment)(nil), ItemType:"", ReplyTo:0, Team:"T24BUPZC1", Reactions:[]slack.ItemReaction(nil)}, SubMessage:(*slack.Msg)(nil)}" module=slack
time="2017-11-13T14:14:47+01:00" level=debug msg="Sending config.Message{Text:"hello from irc", Channel:"#cfelder", Username:"cfelder", UserID:"~cfelder@nat.example.com", Avatar:"", Account:"irc.myirc", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a8491c29c1aaf, ext:26405764907, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)} from irc.myirc (#cfelder) to xmpp.myxmpp (bridge)"
time="2017-11-13T14:14:47+01:00" level=debug msg="Receiving config.Message{Text:"hello from irc", Channel:"bridge", Username:"[irc] <cfelder> ", UserID:"~cfelder@nat.example.com", Avatar:"", Account:"irc.myirc", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a8491c29c1aaf, ext:26405764907, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)}" module=xmpp
<message xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' type='groupchat' xml:lang='en' from='bridge@rooms.example.com/matterbridge'><body>[irc] &lt;cfelder&gt; hello from irc</body></message>
<message xmlns='jabber:client' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' type='groupchat' id='iChat_4109BBF9' from='bridge@rooms.example.com/felder'><body>hello from xmpp</body>
<html xmlns='http://jabber.org/protocol/xhtml-im'><body xmlns='http://www.w3.org/1999/xhtml'>hello from xmpp</body></html></message>
time="2017-11-13T14:14:54+01:00" level=debug msg="Sending message from felder on xmpp.myxmpp to gateway" module=xmpp
time="2017-11-13T14:14:54+01:00" level=debug msg="Sending config.Message{Text:"hello from xmpp", Channel:"bridge", Username:"felder", UserID:"bridge@rooms.example.com/felder", Avatar:"", Account:"xmpp.myxmpp", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a849391706684, ext:33654587976, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)} from xmpp.myxmpp (bridge) to irc.myirc (#cfelder)"
time="2017-11-13T14:14:54+01:00" level=debug msg="Receiving config.Message{Text:"hello from xmpp", Channel:"#cfelder", Username:"[xmpp] <felder> ", UserID:"bridge@rooms.example.com/felder", Avatar:"", Account:"xmpp.myxmpp", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a849391706684, ext:33654587976, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)}" module=irc
time="2017-11-13T14:14:54+01:00" level=debug msg="Sending config.Message{Text:"hello from xmpp", Channel:"bridge", Username:"felder", UserID:"bridge@rooms.example.com/felder", Avatar:"", Account:"xmpp.myxmpp", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a849391706684, ext:33654587976, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)} from xmpp.myxmpp (bridge) to slack.myslack (bridge)"
time="2017-11-13T14:14:54+01:00" level=debug msg="Receiving config.Message{Text:"hello from xmpp", Channel:"bridge", Username:"[xmpp] <felder> ", UserID:"bridge@rooms.example.com/felder", Avatar:"", Account:"xmpp.myxmpp", Event:"", Protocol:"", Gateway:"gateway1", Timestamp:time.Time{wall:0xbe7a849391706684, ext:33654587976, loc:(*time.Location)(0x201a100)}, ID:"", Extra:map[string][]interface {}(nil)}" module=slack
time="2017-11-13T14:14:54+01:00" level=debug msg="Receiving from slackclient &slack.MessageEvent{Msg:slack.Msg{Type:"message", Channel:"C80ARRY15", User:"", Text:"hello from xmpp", Timestamp:"1510578894.000150", ThreadTimestamp:"", IsStarred:false, PinnedTo:[]string(nil), Attachments:[]slack.Attachment{slack.Attachment{Color:"", Fallback:"", CallbackID:"matterbridge", AuthorName:"", AuthorSubname:"", AuthorLink:"", AuthorIcon:"", Title:"", TitleLink:"", Pretext:"", Text:"", ImageURL:"", ThumbURL:"", Fields:[]slack.AttachmentField(nil), Actions:[]slack.AttachmentAction(nil), MarkdownIn:[]string(nil), Footer:"", FooterIcon:"", Ts:""}}, Edited:(*slack.Edited)(nil), SubType:"bot_message", Hidden:false, DeletedTimestamp:"", EventTimestamp:"1510578894.000150", BotID:"B7Z4X73PV", Username:"[xmpp] <felder>", Icons:(*slack.Icon)(nil), Inviter:"", Topic:"", Purpose:"", Name:"", OldName:"", Members:[]string(nil), ReplyCount:0, Replies:[]slack.Reply(nil), ParentUserId:"", File:(*slack.File)(nil), Upload:false, Comment:(*slack.Comment)(nil), ItemType:"", ReplyTo:0, Team:"T24BUPZC1", Reactions:[]slack.ItemReaction(nil)}, SubMessage:(*slack.Msg)(nil)}" module=slack
<iq xmlns='jabber:client' type='get' to='felder@example.com/a8a8fcd78fa9fa85772ee1515e07b2d1e5b841a0' from='rooms.example.com' id='DewALQ6aTBG4C0wBUMy52g=='><ping xmlns='urn:xmpp:ping'/></iq>
```

### Any steps to reproduce the behavior?

* configure a (inout) gateway between slack and either irc or xmpp.

#### (be sure to exclude or anonymize private data (tokens/passwords))

```
[slack]
    [slack.myslack]
    #Token to connect with the Slack API
    #You'll have to use a test/api-token using a dedicated user and not a bot token.
    #See https://github.com/42wim/matterbridge/wiki/Slack-bot-setup for more information
    Token="changeme-token"
    RemoteNickFormat="[{PROTOCOL}] <{NICK}> "

[irc]
    [irc.myirc]
    Server="irc.opensuse.org:6667"
    Nick="mattermost"
    RemoteNickFormat="[{PROTOCOL}] <{NICK}> "

[xmpp]
    [xmpp.myxmpp]
    Server="example.com:5222"
    #Jid your userid
    Jid="felder@example.com"
    Password="changeme"
    Muc="rooms.example.com"
    Nick="matterbridge"
    RemoteNickFormat="[{PROTOCOL}] <{NICK}> "


[[gateway]]
name="gateway1"
enable=true

    [[gateway.inout]]
    account="irc.myirc"
    #channel="#opensuse-buildservice"
    channel="#cfelder"

    [[gateway.inout]]
    account="slack.myslack"
    channel="bridge"

    [[gateway.inout]]
    account="xmpp.myxmpp"
    channel="bridge"
```

**Comment by 42wim** [comment]
Is it possible you build from master ?
Should be fixed now (if you rebuild from master)

If you didn't build from master something else is wrong :)

**Comment by cfelder** [comment]
I just followed the instructions from your Readme using go get which probably installs the latest release. I will try to rebuild from master when I am back in office tomorrow. Can you please point me into the right direction how to do this. I am new to go and I haven't seen a Makefile which invokes the go compiler.

By the way thanks for your fast response :)

**Comment by 42wim** [comment]
I just released v1.4.1 binaries (https://github.com/42wim/matterbridge/releases/tag/v1.4.1) which should fix your issue :)

Or if you want to build yourself again, you can do by running a `go get -u github.com/42wim/matterbridge` the `-u` make sure it updates to latest master.

I try to keep away from Makefiles unless really needed.

**Comment by cfelder** [comment]
Works as expected. Thank you very much for the immediate fix.

I tested both v1.4.1 and 1.4.2-dev but from the git log I expect they are basically the same at the moment.
